hera_mc/cm_hookup.py

```diff
@@ -346,60 +54,26 @@ def delete_cache_file(self):
         if os.path.exists(self.hookup_cache_file):
             os.remove(self.hookup_cache_file)
 
-    def _hookup_cache_to_use(self, force_new_cache=False):
+    def get_hookup(self, hpn, pol='all', at_date='now', exact_match=False,
+                   use_cache=False, hookup_type='parts_hera'):
         """
-        Confirms/sets the cached_hookup_dict to use.  If force_new_cache is set, it automatically
-        writes/uses a new one.  If the cache file is out of date it writes/uses a new one.  Otherwise
-        it uses the existing one (rereading from the cache file if not one in memory.)
-
-        If the cache file is up-to-date and is in memory, this does nothing.
-
-        Parameters
-        ----------
-        force_new_cache : bool
-            Flag to force a new cache hookup to be set and written.
-        """
-        if force_new_cache or not self._hookup_cache_file_OK():
-            self.cached_hookup_dict = self.get_hookup_from_db(hpn_list=self.hookup_list_to_cache, rev='ACTIVE',
-                                                              port_query='all', at_date=self.at_date,
-                                                              exact_match=False, hookup_type=self.hookup_type)
-            log_msg = "force_new_cache:  {}".format(force_new_cache)
-            self.write_hookup_cache_to_file(log_msg)
-        elif self.cached_hookup_dict is None:
-            if os.path.exists(self.hookup_cache_file):
-                self.read_hookup_cache_from_file()
-            else:  # pragma: no cover
-                self._hookup_cache_to_use(force_new_cache=True)
-
-    def get_hookup(self, hpn_list, rev='ACTIVE', port_query='all', at_date='now', exact_match=False,
-                   force_new_cache=False, force_db=False, hookup_type=None):
-        """
-        Return the full hookup to the supplied part/rev/port in the form of a dictionary.
-        The data may come from one of two sources:
-        (1) if the (a) part list is consistent with the keys, (b) date is current, and (c) revision
-            hash is current with a local cache file, data from the cache file will be used.
-        (2) if those conditions are not met, or the flag 'force_new_cache' is True, or 'force_db'
-            is True, it will use the database.
+        Return the full hookup to the supplied part/rev/pol in the form of a dictionary. It will
+        return all active revisions at_date from either the database or the cache file if
+        use_cache == True and the part number keys agree.
```

Comment:
The first line in a docstring should be a single short sentence that fits on one line. Then there should be an empty line before any longer descriptions.

Reply:
Done